app/src/main/java/net/gsantner/markor/util/StringUtils.java

```diff
@@ -0,0 +1,67 @@
+package net.gsantner.markor.util;
+
+import org.apache.commons.lang3.ArrayUtils;
+
+public class StringUtils extends org.apache.commons.lang3.StringUtils {
+    // modified version of org.apache.commons.lang3.StringUtils#indexOfAny
+    /**
+     * Same as {@link org.apache.commons.lang3.StringUtils#indexOfAny(CharSequence, char...)
+     * where you can specify the search intervall}
+     */
+    public static int indexOfAny(final CharSequence cs, int csFirst, int csLen, final char... searchChars) {
+        if (isEmpty(cs) || ArrayUtils.isEmpty(searchChars)) {
+            return INDEX_NOT_FOUND;
+        }
+        final int csLast = csLen - 1;
+        final int searchLen = searchChars.length;
+        final int searchLast = searchLen - 1;
+        for (int i = csFirst; i < csLen; i++) {
+            final char ch = cs.charAt(i);
+            for (int j = 0; j < searchLen; j++) {
+                if (searchChars[j] == ch) {
+                    if (i < csLast && j < searchLast && Character.isHighSurrogate(ch)) {
+                        // ch is a supplementary character
+                        if (searchChars[j + 1] == cs.charAt(i + 1)) {
+                            return i;
+                        }
+                    } else {
+                        return i;
+                    }
+                }
+            }
+        }
+        return INDEX_NOT_FOUND;
+    }
+
+    /**
+     * returns search for begin of line starting for csFirst down to 0
+     */
+    public static int beginOfLine(final CharSequence cs, int csFirst) {
```

Comment:
https://github.com/gsantner/markor/blob/master/app/src/main/java/net/gsantner/markor/frontend/textview/TextViewUtils.java#L67

We already have this

Reply:
Moved from [net.gsantner.markor.util.StringUtils] to [net.gsantner.opoc.format.GsTextUtils]. Replaced implementation with getNeighbourLineEndings() . Created unittests for getNeighbourLineEndings() to find out what it exactly does and updated docs